woocommerce/woocommerce-gutenberg-products-block
# Add Payment Methods Registry for registering scripts and data

Adds a payment method registry so payment integrations can use a common interface. Modelled like Block registration.

Payment classes on the PHP side register like this:

```php
PaymentMethodRegistry::get_instance()->register(
	$this->payment_method_name,
	array(
		'scripts' => $this->get_payment_method_scripts(),
		'data'    => $this->get_payment_method_script_data(),
	)
);
```

`data` contains the key value pairs to add to the settings array. The key of the data is set to `GATEWAY_data`.

`scripts` are payment scripts/handles which get passed to the Assets class for registration.

The `PaymentMethodAssets` class takes registered payment methods and enqueues their scripts and data when needed. They are enqueued in this case when the checkout is rendered.

**Additional:**
- Using this new system, I also added a **Cheque** gateway which does little processing, just processes orders.
- Each payment method integration needs its own handle so I updated webpack config to split them (stripe and cheque)
- I refactored some of the apple pay/stripe code and directory structure to better organise things

Fixes #2067

### How to test the changes in this Pull Request:

Build then go to checkout and confirm you see Stripe (if setup) and cheque. You may need to run `composer dump-autoload` first.
